cmd/terramate/cli/cli.go

```diff
@@ -2112,15 +2113,18 @@ func (c *cli) friendlyFmtDir(dir string) (string, bool) {
 	return prj.FriendlyFmtDir(c.rootdir(), c.wd(), dir)
 }
 
-func (c *cli) computeSelectedStacks(ensureCleanRepo bool, cloudStatus cloudstack.FilterStatus) (config.List[*config.SortableStack], error) {
+func (c *cli) computeSelectedStacks(filterStacks bool, ensureCleanRepo bool, cloudStatus cloudstack.FilterStatus) (config.List[*config.SortableStack], error) {
 	report, err := c.listStacks(c.parsedArgs.Changed, cloudStatus)
 	if err != nil {
 		return nil, err
 	}
 
 	c.gitFileSafeguards(ensureCleanRepo)
 
-	entries := c.filterStacks(report.Stacks)
+	entries := report.Stacks
+	if filterStacks {
+		entries = c.filterStacks(report.Stacks)
+	}
```

Comment:
I see that you want to compute the selected stacks twice, first filtering (by changed, tags, workingdir, cloud status, etc) and then later without some filters but allowing others (changed, cloud status). This seems broken.

Additionally, this logic requires a lot of computation, invoking of lots of git commands, parsing of Terraform modules, etc, then doing this twice is very bad.